src/apphosting/secrets/index.ts

```diff
@@ -164,3 +168,115 @@ export async function upsertSecret(
   // This may not be a great idea until version control is actually implemented.
   return false;
 }
+
+/**
+ * Fetches secrets from Google Secret Manager and returns their values in plain text.
+ *
+ * This function takes an array of `Secret` objects, each containing the name
+ * and URL of a secret stored in Google Secret Manager. It retrieves the
+ * secrets from Secret Manager and returns a Map where the keys are the secret
+ * names and the values are the corresponding secret values in plain text.
+ *
+ * @param projectId The ID of the Google Cloud project where the secrets are stored.
+ * @param secrets An array of `Secret` objects, each specifying the name and URL of a secret.
+ * @returns A Promise that resolves to a Map containing the secret names as keys and their
+ *          plain text values as values.
+ */
+export async function fetchSecrets(
+  projectId: string,
+  secrets: Secret[],
+): Promise<Map<string, string>> {
+  const secretsKeyValuePairs: Map<string, string> = new Map();
+
+  try {
+    for (const secretConfig of secrets) {
+      let [name, version] = secretConfig.secret!.split("@");
+      if (!version) {
+        version = "latest";
+      }
+
+      const value = await gcsm.accessSecretVersion(projectId, name, version);
+      secretsKeyValuePairs.set(secretConfig.variable, value);
+    }
+  } catch (e) {
+    throw new FirebaseError(`Error exporting secrets: ${e}`);
+  }
+
+  return secretsKeyValuePairs;
+}
+
+/**
+ * Determines the appropriate App Hosting YAML configuration for exporting secrets.
+ *
+ * When exporting secrets, users need to select an `apphosting.*.yaml` file that corresponds
+ * to the environment they want to export from.
+ *
+ * Environment-specific configurations are prioritized over the default `apphosting.yaml`
+ * to ensure the exported secrets accurately reflect the chosen environment.
+ *
+ * @param allAppHostingYamlPaths An array of paths to all available `apphosting.*.yaml` files.
+ * @returns A Promise that resolves to an `AppHostingYamlConfig` object representing the combined
+ *          configuration to be used for exporting configurations.
+ */
+export async function getConfigToExport(
+  allAppHostingYamlPaths: string[],
+): Promise<AppHostingYamlConfig> {
+  const fileNameToPathMap: Map<string, string> = new Map();
+  for (const path of allAppHostingYamlPaths) {
+    const fileName = basename(path);
+    fileNameToPathMap.set(fileName, path);
+  }
+  const baseFilePath = fileNameToPathMap.get(APPHOSTING_BASE_YAML_FILE);
+  const appHostingfileToExportPath = await promptForAppHostingYaml(fileNameToPathMap);
```

Comment:
we need to account for the case where the user specified `export --secrets apphosting.staging.yaml`

Reply:
updated: implemented logic for this.